src/app/modules/ServiceAccounts/components/ServiceAccountsTableSection.tsx

```diff
@@ -21,15 +20,20 @@ export const ServiceAccountsTableSection: React.FunctionComponent<
     useModal<ModalType.KasCreateServiceAccount>();
   const { showModal: showDeleteServiceAccountModal } =
     useModal<ModalType.KasDeleteServiceAccount>();
-  const location = useLocation();
 
   const [orderBy, setOrderBy] = useState<string>('name asc');
-  const [filterSelected, setFilterSelected] = useState('name');
-  const [filteredValue, setFilteredValue] = useState<FilterType[]>([]);
 
-  const searchParams = new URLSearchParams(location.search);
   const { page = 1, perPage = 10 } = usePagination() || {};
-  const mainToggle = searchParams.has('user-testing');
+
+  const [tableData, setTableData] =
+    useState<ServiceAccountListItem[]>(serviceAccountItems);
+  const [filteredData, setFilteredData] =
+    useState<ServiceAccountListItem[]>(serviceAccountItems);
+
+  useEffect(() => {
+    setTableData(tableData);
+    setFilteredData(tableData);
+  }, []);
```

Comment:
What's this useEffect supposed to do? I'm trying to unwrap it in my head but I think this does nothing. What am I missing?